src/XMakeTasks/NativeMethods.cs

```diff
@@ -798,6 +804,32 @@ internal static bool MakeHardLink(string newFileName, string exitingFileName, ou
         }
 
         //------------------------------------------------------------------------------
+        // CreateSymbolicLink
+        //------------------------------------------------------------------------------
+        [DllImport("kernel32.dll", SetLastError = true, CharSet = CharSet.Unicode)]
+        internal static extern bool CreateSymbolicLink(string symLinkFileName, string targetFileName, SymbolicLink dwFlags);
+
+        [DllImport("libc", SetLastError = true)]
+        internal static extern int symlink(string oldpath, string newpath);
+
+        internal static bool MakeSymbolicLink(string newFileName, string exitingFileName, ref string errorMessage)
+        {
+            bool symbolicLinkCreated;
+            if (NativeMethodsShared.IsWindows)
+            {
+                symbolicLinkCreated = CreateSymbolicLink(newFileName, exitingFileName, SymbolicLink.File);
+                errorMessage = symbolicLinkCreated ? null : Marshal.GetExceptionForHR(Marshal.GetHRForLastWin32Error()).Message;
+            }
+            else
+            {
+                symbolicLinkCreated = symlink(exitingFileName, newFileName) == 0;
+                errorMessage = symbolicLinkCreated ? null : "The link() library call failed with the following error code: " + Marshal.GetLastWin32Error();
```

Comment:
In case anyone else has the same question I did: [yes](https://github.com/dotnet/corefx/issues/6778#issuecomment-199027445), `Marshal.GetLastWin32Error` returns `errno`.